File: leetcode/118-pascals-triangle.py
# https://leetcode.com/problems/pascals-triangle/ (top-interview-questions)

from typing import List

class Solution:
    def generate(self, numRows: int) -> List[List[int]]:
        numRowsArray = [[1]]

        for rowNum in range(1, numRows):
            if rowNum == 1: numRowsArray.append([1, 1])
            else:
                numRowsArray.append([1])
                for i in range(1, len(numRowsArray[rowNum-1])):
                    sum = numRowsArray[rowNum-1][i-1] + numRowsArray[rowNum-1][i]
                    numRowsArray[rowNum].append(sum)
                numRowsArray[rowNum].append(1)

        return numRowsArray

File: leetcode/118-pascals-triangle.js
// https://leetcode.com/problems/pascals-triangle/ (top-interview-questions)

/**
 * @param {number} numRows
 * @return {number[][]}
 */
var generate = function(numRows) {
  const rows = [[1]];

  for (let i = 1; i < numRows; i++) {
    const row = new Array(i+1).fill(1);

    for (let j = 1; j < i; j++) {
      row[j] = rows[i-1][j-1] + rows[i-1][j];
    }
    rows.push(row);
  }

  return rows;
};
